src/db.c

```diff
@@ -727,7 +734,7 @@ void delGenericCommand(client *c, int lazy) {
     int numdel = 0, j;
 
     for (j = 1; j < c->argc; j++) {
-        expireIfNeeded(c->db,c->argv[j],0);
+        if (expireIfNeeded(c->db,c->argv[j],0) == KEY_DELETED) continue;
```

Comment:
```suggestion
        if (expireIfNeeded(c->db,c->argv[j],0) == KEY_DELETED)
            continue;
```
i usually prefer to put statements that change the flow in their own line (not together with the if).
that is unless it's error handling pattern that's repeated a ton of times.